app/models/cart/validator.rb

```diff
@@ -0,0 +1,25 @@
+module Shop
+  module Models
+    class Cart
+      class Validator
+        def initialize(cart)
+          @cart = cart
+        end
+
+        def validate!
+          raise ArgumentError if @cart.items.nil?
+          raise ArgumentError unless cart_items_only?
+        end
+
+        private
+
+        def cart_items_only?
+          @cart.items.each do |item|
+            return false unless item.is_a?(CartItem)
+          end
+          true
```

Comment:
I prefer code using exception, like

```ruby
 def cart_items_only?
  raise XXX unless @cart.items.all?{|x| x.is_a?(CartItem)} 
end
```
much cleaner, and you can catch those exception higher and do sth